### ISSUE TITLE
Issue/Question: Command failed: convert.im6: delegate failed html2ps + no image defined

### ISSUE THREAD

**Issue by sebastienratez** [ImageMagick/ImageMagick#398]
This is a duplicate of corresponding gm issue - https://github.com/aheckmann/gm/issues/644
But since the subclass is ImageMagick, I think it is in fact more suited here.

So, I'm not too sure what's happening here. I'm running imagemagick through the npn gm module in nodejs (v6):

```

var gm = require('gm').subClass({
  imageMagick: true
});
var fs = require('fs');

var file = "/absolutepath/dis.png";
var targetFile = "/absolutepath/300px-dis.png";

if (!fs.existsSync(file)) console.log("WARNING:  FILE DOES NOT EXIST");
gm(file).resize(300).write(targetFile, function(err) {
              if (err) {
                console.log("[KO - RESIZE 300]", err);
                return callback(err);
              }
});
```

The command fails with the following error message:

```
	[KO - RESIZE 300] Error: Command failed: convert.im6: delegate failed `\"html2ps\" -U -o \"%o\" \"%i\"' @ error/delegate.c/InvokeDelegate/1065.,
    convert.im6: unable to open image `/tmp/magick-dEhsHcmz': No such file or directory @ error/blob.c/OpenBlob/2641.,
    convert.im6: unable to open file `/tmp/magick-dEhsHcmz': No such file or directory @ error/constitute.c/ReadImage/583.,
    convert.im6: no images defined `/absolutepath/300px-dis.png' @ error/convert.c/ConvertImageCommand/3044.,,
    at ChildProcess.onExit (/nodejsAction/node_modules/gm/lib/command.js:301:17),
```

What troubles me is that I'm pretty sure this code used to work a while ago.

 - The other thing is I don't really know why html2ps should be involved as `file` is a png file.
 - I don't get the warning message in the console so it means `file` does exists.
 - I am using absolute paths, which I make sure they exist earlier in the process

I really don't have a clue where and how to proceed so I was wondering if you could point me out to the right direction.

Thx

**Comment by sebastienratez** [comment]
I tried with a jpg image and the problem remains. Why is it calling html2ps? I mean I don't know if it's the issue but it does not really seem right

**Comment by mikayla-grace** [comment]
ImageMagick utilizes delegate libraries or programs to read or write a select number of formats-- in the case of an HTML file, it leverages the html2ps program.  Now you claim the file is PNG, however, ImageMagick is recognizing the first few bytes of the image as HTML.  To debug, you can set the MAGICK_DEBUG environment variable to 'coder' to gain some insight.  If that fails, try 'trace'.  Trace returns a huge amount of data-- good luck sifting through it.

Also make sure PNG is supported in your version of ImageMagick.  Type

```
convert -list format
```

**Comment by sebastienratez** [comment]
I converted the PNG to JPG using the usual windows editor tool, then tried imagemagick on it, and it did the same thing. I'll see what the magick_debug variable does, thanks

**Comment by sdkevorkian** [comment]
Hello! I just wanted to chime in and say that I had my program set up working fine locally, but when I deployed it to Heroku I got this error. I am not trying to convert intentionally, I am downloading 2 jpgs, then resizing them, using compare, and then deleting them. 

I hadn't touched my code since the middle of june, and it had previously worked on Heroku, but today I just wanted to add analytics and clean up some comments/console.logs and I had this error that prevented my program from working. I was able to rollback to a previous deployment for now but I am not sure what changed? I did update node (version 6.11.0) in the time I started my project and ran "npm rebuild" before deploying today.
